### PR TITLE
 fixed:issue->#1968 Import from nYNAB fails with unknown error

### PR DESCRIPTION
**Pull request by Marethyu1**
Allow case insensitive ynab5 import for special 'starting balance' payee

Fixes issue for attached budget in this https://github.com/actualbudget/actual/issues/1968#issuecomment-1880111861 
by allow case insensitive search for starting balance. 

Should address #1968 as described in this comment https://github.com/actualbudget/actual/issues/1968#issuecomment-1880192146

### CODE REVIEW

**Comment by MatissJanis** [comment]
Thanks! Would you be OK adding the same logic for the other category/payee matchers in the same file? Might as well fix it everywhere if we've gone so far.

File: packages/loot-core/src/server/importers/ynab5.ts [hunk 1]
```diff
@@ -136,8 +136,12 @@ async function importTransactions(
   const startingBalanceCatId = categories.find(
     cat => cat.name === 'Starting Balances',
   ).id; //better way to do it?
+
   const startingPayeeYNAB = data.payees.find(
-    payee => payee.name === 'Starting Balance',
+    payee =>
+      payee.name.localeCompare('Starting Balance', undefined, {
```
Comment (joel-jeremy) [inline packages/loot-core/src/server/importers/ynab5.ts:142]: Can we create a `equalsIgnoreCase` function here? And we also do an ignore case equality check on `Starting Balances` category just above this line?

**Review by joel-jeremy — CHANGES_REQUESTED** [review]

**Comment by Marethyu1** [comment]
👋 I have extracted the check and applied it to all the string literals in the file. Seems like it might be out of the scope for what we initially set out to change. Let me know what y'all think, can always drop the last couple of commits. Thanks.

File: packages/loot-core/src/server/importers/ynab5.ts (hunk 1 above)
Comment (Marethyu1) [reply packages/loot-core/src/server/importers/ynab5.ts]: Done this, and also applied to to other string literals in the file

**Review by MatissJanis — DISMISSED** [review]
Thanks!

File: packages/loot-core/src/server/importers/ynab5.ts [hunk 2]
```diff
@@ -327,3 +327,19 @@ export function parseFile(buffer: Buffer): YNAB5.Budget {
...
+    }) === 0
+  );
+}
+
+function findByNameIgnoreCase(categories: YNAB5.CategoryGroup[], name: string) {
+  return categories.find(cat => equalsIgnoreCase(cat.name, name));
+}
+
+function findIdByName(categories: YNAB5.CategoryGroup[], name: string) {
+  return findByNameIgnoreCase(categories, name).id;
```
Comment (joel-jeremy) [inline packages/loot-core/src/server/importers/ynab5.ts:344]: We need to handle for null here:
```suggestion
  return findByNameIgnoreCase(categories, name)?.id;
```

**Review by joel-jeremy — APPROVED** [review]
Thanks!

**Pull request merged**